core/chains/evm/gas/l2_suggested_estimator.go

```diff
@@ -0,0 +1,164 @@
+package gas
+
+import (
+	"context"
+	"math/big"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/pkg/errors"
+
+	evmtypes "github.com/smartcontractkit/chainlink/core/chains/evm/types"
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/utils"
+)
+
+var (
+	_ Estimator = &l2SuggestedEstimator{}
+)
+
+//go:generate mockery --name rpcClient --output ./mocks/ --case=underscore --structname OptimismRPCClient
+type rpcClient interface {
+	Call(result interface{}, method string, args ...interface{}) error
+}
+
+type l2SuggestedEstimator struct {
+	utils.StartStopOnce
+
+	config     Config
+	client     rpcClient
+	pollPeriod time.Duration
+	logger     logger.Logger
+
+	gasPriceMu sync.RWMutex
+	l2GasPrice *big.Int
+
+	chForceRefetch chan (chan struct{})
+	chInitialised  chan struct{}
+	chStop         chan struct{}
+	chDone         chan struct{}
+}
+
+// NewL2SuggestedEstimator returns a new optimism 2.0 estimator
+func NewL2SuggestedEstimator(lggr logger.Logger, config Config, client rpcClient) Estimator {
+	return &l2SuggestedEstimator{
+		utils.StartStopOnce{},
+		config,
+		client,
+		10 * time.Second,
+		lggr.Named("L2SuggestedEstimator"),
+		sync.RWMutex{},
+		nil,
+		make(chan (chan struct{})),
+		make(chan struct{}),
+		make(chan struct{}),
+		make(chan struct{}),
+	}
+}
+
+func (o *l2SuggestedEstimator) Start(context.Context) error {
+	return o.StartOnce("L2SuggestedEstimator", func() error {
+		go o.run()
+		<-o.chInitialised
+		return nil
```

Comment:
Shouldn't we check that o.l2GasPrice != nil before returning success here?

If for some reason the rpcClient is unreachable, doesn't that make this estimator useless?